ScoreSaber/Core/Services/PlayerService.cs

```diff
@@ -33,162 +37,101 @@ public void ChangeLoginStatus(LoginStatus _loginStatus, string status) {
-                    platform = "1";
-                    break;
+            var authToken = await platformUserModel.GetUserAuthToken();
+            var (nonce, platform) = userInfo.platform switch {
+                UserInfo.Platform.Steam => (authToken.token, "0"),
+                UserInfo.Platform.Oculus => ($"{authToken.token},{(await platformUserModel.RequestXPlatformAccessToken(CancellationToken.None)).token}", "1"),
+                _ => (string.Empty, string.Empty)
+            };
+            if (string.IsNullOrEmpty(nonce) || string.IsNullOrEmpty(platform)) {
+                ChangeLoginStatus(LoginStatus.Error, "Failed to authenticate with ScoreSaber: Invalid platform");
+                return;
             }
-
-            var playerId = userInfo.platformUserId;
-            var playerName = userInfo.userName;
             var friendIds = await platformUserModel.GetUserFriendsUserIds(false);
-            var friends = string.Join(",", friendIds.Where(x => x != "0"));
-
-            var playerInfo = new LocalPlayerInfo(playerId, playerName, friends, platform, nonce);
-
-            int attempts = 1;
-
-            while (attempts < 4) {
-
-                var authenticated = await AuthenticateWithScoreSaber(playerInfo);
-
-                if (authenticated) {
+            var playerInfo = new LocalPlayerInfo(
+                userInfo.platformUserId,
+                userInfo.userName,
+                string.Join(",", friendIds.Where(x => x != "0")),
+                platform,
+                nonce
+            );
+            for (int attempt = 1; attempt <= 3; attempt++) {
+                if (await AuthenticateWithScoreSaber(playerInfo)) {
                     localPlayerInfo = playerInfo;
-                    string successText = "Successfully signed into ScoreSaber!";
-                    if (localPlayerInfo.playerId == PlayerIDs.Denyah)
-                        successText = "Wagwan piffting wots ur bbm pin?";
-
+                    var successText = localPlayerInfo.playerId == PlayerIDs.Denyah
+                        ? "Wagwan piffting wots ur bbm pin?"
+                        : "Successfully signed into ScoreSaber!";
                     ChangeLoginStatus(LoginStatus.Success, successText);
-                    break;
-                } else {
-                    ChangeLoginStatus(LoginStatus.Error, $"Failed, attempting again ({attempts} of 3 tries...)");
-                    attempts++;
-                    await Task.Delay(4000);
+                    return;
                 }
-
-            }
-
-            if (loginStatus != LoginStatus.Success) {
-                ChangeLoginStatus(LoginStatus.Error, "Failed to authenticate with ScoreSaber! Please restart your game");
+                ChangeLoginStatus(LoginStatus.Error, $"Failed, attempting again ({attempt} of 3 tries...)");
+                await Task.Delay(4000);
```

Comment:
Isn't this also being shown after the 3rd attempt too with a wait of 4s until it shows you that it failed without every trying again?

Reply:
fixed